qds_sdk/connection.py

```diff
@@ -188,15 +188,18 @@ def _handle_error(response):
         elif code == 422:
             sys.stderr.write(response.text + "\n")
             raise ResourceInvalid(response)
-        elif code in (502, 503, 504):
+        elif code in (502, 504):
             sys.stderr.write(response.text + "\n")
             raise RetryWithDelay(response)
         elif code == 449:
             sys.stderr.write(response.text + "\n")
             raise RetryWithDelay(response, "Data requested is unavailable. Retrying...")
-        elif code == 429:
+        elif code in (429, 503):
             sys.stderr.write(response.text + "\n")
-            raise ApiThrottledRetry(response, "Too many requests. Retrying...")
+            if code == 429
+              raise ApiThrottledRetry(response, "Too many requests. Retrying...")
+            else
+              raise ApiThrottledRetry(response, "Service Unavailable. Retrying...")
```

Comment:
Define a common method `get_error_message` to get the error message
```suggestion
            raise ApiThrottledRetry(get_error_message(code))
```